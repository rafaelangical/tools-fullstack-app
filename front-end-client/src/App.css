.App {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  width: 100%;
  height: 100%;
}
.container{
  width: 80%;
  height: 100%;
  margin: 0 auto;
}
.div-top-title{
  width: 100%;
  height: 30%;
}
.div-top-title > h1 {
  font-size: 3em;
  font-weight: 250;
  letter-spacing: 0.5ch;
}
.div-top-title > h3 {
  font-size: 1.6em;
  font-weight: 250;
}
.div-search-add {
  width: 100%;
  height: 10%;
  display: flex;
  flex-flow: row wrap;
}

.div-search-add-left, .div-search-add-right{
  width: 50%;
  height: 100%;
}

.div-search-add-right{
  display: flex;
  justify-content: end;
}

.div-search-add-left > input.search {
  width: 50%;
}

.section-main{
  width:100%;
  height: 70%;
  display: flex;
  flex-flow: row wrap;
  margin: 0 auto;
}

.div-tools {
  width: 100%;
  height: 30%;
  padding: 1em;
  margin-top: 1em;
}

.div-tools-top{
  width: 100%;
  height: 10%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  font-size: 1.7em;
}

.div-tools-top > a{
  color:  dodgerblue; 
}

p.tags{
  font-size: 1.2em;
  font-weight: 40000;
  color: black;
}