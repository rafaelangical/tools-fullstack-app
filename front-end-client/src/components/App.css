.App {
  display: flex;
  flex-flow: row wrap;
  align-items: flex-start;
  width: 100%;
  max-height: 100%;
  overflow: auto;
}
.container{
  width: 80vw;
  height: 100%;
  margin: 0 auto;
}
.div-top-title{
  width: 100%;
  height: 10%;
  display: flex;
  flex-flow: column wrap;
}
.div-top-title > h1 {
  font-size: 3em;
  font-weight: 250;
  letter-spacing: 0.5ch;
}
.div-top-title > h3 {
  font-size: 1.6em;
  font-weight: 250;
}
.div-search-add {
  width: 100%;
  height: 5%;
  display: flex;
  flex-flow: row wrap;
}

.div-search-add-left, .div-search-add-right{
  width: 50%;
  height: 100%;
}

.div-search-add-right{
  display: flex;
  justify-content: end;
}

.div-search-add-left > input.search {
  width: 50%;
}

.section-main{
  width:100%;
  height: 70%;
  display: flex;
  flex-flow: row wrap;
  margin: 0 auto;
}

.div-tools {
  width: 100%;
  height: 30%;
  padding: 1em;
  margin-top: 1em;
}

.div-tools-top{
  width: 100%;
  height: 10%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  font-size: 1.7em;
}

.div-tools-top > a{
  color:  dodgerblue; 
}

p.tags{
  font-size: 1.2em;
  font-weight: 40000;
  color: black;
}

.button-add{
  color: white;
  border: 1px solid black;
  background-color: blue;
  width: 15%;
  height: 2em;
  font-size: 1em;
  border-radius: 0.5em;
}

.button-remove{
  color: black;
  border: 1px solid #999;
  background-color: darkred; 
  border-radius: 0.5em;
}