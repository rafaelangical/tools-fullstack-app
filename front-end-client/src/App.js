import React, { Component } from 'react';
import './App.css';
import axios from 'axios';
import { join } from 'upath';

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      response: [],
      data: []
    }

    this.renderData = this.renderData.bind(this);
  }

  componentDidMount() {
    axios.get('http://localhost:3000/tools')
      .then(res => {
        this.setState({ data: res.data})
        console.log(res.data)
      })
      .catch(err => console.log(err))
  }

  renderData() {
    const data = this.state.data;
    data.map((item,index)=> {
      return <div key={index}>
        <p>{item}</p>
        <h3>{item.description}</h3>
        {item.tags.map((item,index) => {
          return <ul key={index}>
            <li>{item}</li>
          </ul>
        })}
      </div>
    })
  }

  render() {
    const data = this.state.data;
    console.log(this.state.data);

    return (
      <div className="App">
        <main className="container">
          <section className="div-top-title">
            <h1>VUTTR</h1>
            <h3>Very Useful Tools to Remember</h3>
          </section>
          <section className="div-search-add">
            <div className="div-search-add-left">
              <input type="text" placeholder="search" className="search"/>
              <input type="radio"/><label>search in tags only</label>
            </div>
            <div className="div-search-add-right">
              <button>+ add</button>
            </div>
          </section>
          <section className="section-main">
              {data.map((item,index, data)=> {
                return <div className="div-tools" key={index} style={{border: "1px solid black"}}>
                  <div className="div-tools-top">
                    <a href="#">{item.title}</a>
                    <button>X remove</button>
                  </div>
                  <p>{item.description}</p>
                  {function() {
                    const tag = item.tags;
                    const newarray = new Array(tag).join('');
                    console.log(newarray);
                    return <p className="tags">Tags: {newarray}</p>
                  }()}
                </div>
              })
              }
          </section>
        </main>
      </div>
    )
  }
}

export default App;
