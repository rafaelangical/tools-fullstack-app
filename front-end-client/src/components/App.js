import React, { Component, Fragment } from 'react';
import './App.css';
import axios from 'axios';
import AddModal from './addModal.js';
import RemoveModal from './removeModal.js';

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      data: [],
      search: '',
      id: 0,
      withTag: false,
      isOpen: false,
      isOpenAdd: false
    }
    this.closeModal = this.closeModal.bind(this);
    this.closeModalAdd = this.closeModalAdd.bind(this);
  }

  componentDidMount() {
    axios.get('http://localhost:3000/tools')
      .then(res => {
        this.setState({ data: res.data})
      })
      .catch(err => console.log(err))
  }

  remove(index) {
    this.setState({id: index});
    this.setState({isOpen: !this.state.isOpen})
  }

  closeModal() {
    this.setState({isOpen: !this.state.isOpen});
    this.componentDidMount();
  }
  
  closeModalAdd() {
    this.setState({isOpenAdd: !this.state.isOpenAdd});
    this.componentDidMount();
  }


  handleSearch(e) {
    if(e.length<1){
      this.componentDidMount();
    }
    this.setState({search: e});
    const search = this.state.search;
    if(this.state.withTag){
      axios.get(`http://localhost:3000/tools?tags_like=${search}`)
      .then(res => {
        this.setState({ data: res.data})
      })
      .catch(err => console.log(err))
    }else{
      axios.get(`http://localhost:3000/tools?q=${search}`)
      .then(res => {
        this.setState({ data: res.data})
      })
      .catch(err => console.log(err))
    }
  }

  handleSearchWithTags(e) {
    this.setState({search: e});
    const search = this.state.search;
    axios.get(`http://localhost:3000/tools?tags_like=${search}`)
      .then(res => {
        this.setState({ data: res.data})
        console.log(res.data)
      })
      .catch(err => console.log(err))
  }

  render() {
    const data = this.state.data;
    return (
      <Fragment>
      <div className="App">
        <main className="container">
          <section className="div-top-title">
            <h1>VUTTR</h1>
            <h3>Very Useful Tools to Remember</h3>
          </section>
          <section className="div-search-add">
            <div className="div-search-add-left">
              <input type="text" placeholder="search" className="search" onChange={(e) => this.handleSearch(e.target.value)} value={this.state.search}/>
              <input type="checkbox" onChange={() => this.setState({withTag: true})}/>
              <label>search in tags only</label>
            </div>
            <div className="div-search-add-right">
              <button onClick={() => this.setState({isOpenAdd: !this.state.isOpenAdd})}>+ add</button>
            </div>
          </section>
          <section className="section-main">
              {data.map((item, data)=> {
                return <div className="div-tools"key={item.id} style={{border: "1px solid black"}}>
                  <div className="div-tools-top">
                    <a href={item.id}>{item.title}</a>
                    <button onClick={() => this.remove(item.id)}>X remove</button>
                  </div>
                  <p>{item.description}</p>
                  {function() {
                    const tag = item.tags;
                    const newarray = new Array(tag).join("");
                    return <p className="tags">Tags: {newarray}</p>
                  }()}
                </div>
              })
              }
          </section>
        </main>
        <AddModal 
          show={this.state.isOpenAdd}
          closeModalAdd={this.closeModalAdd}
        />
        <RemoveModal
          show={this.state.isOpen}
          closeModal={this.closeModal}
          id={this.state.id}
        /> 
        </div>
      </Fragment>
    )
  }
}

export default App;
