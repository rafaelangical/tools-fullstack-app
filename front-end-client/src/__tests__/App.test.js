import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() })

describe('Test if App is rendered correctly', () => {
  test('renders correctly App', () => {
    const tree = renderer.create(
      <App />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test a state of App', () => {
  test('have data state', () => {
    const wrapper = shallow(<App />);
    const dataState = wrapper.state('data');
    expect(dataState).toBeDefined();
  });
  
  test('have search state', () => {
    const wrapper = shallow(<App />);
    const searchState = wrapper.state('search');
    expect(searchState).toBeDefined();
  });

  test('have id state', () => {
    const wrapper = shallow(<App />);
    const idState = wrapper.state('id');
    expect(idState).toBeDefined();
  });

  test('have withTag state', () => {
    const wrapper = shallow(<App />);
    const withTagState = wrapper.state('withTag');
    expect(withTagState).toBeDefined();
  });

  test('have isOpen state', () => {
    const wrapper = shallow(<App />);
    const isOpenState = wrapper.state('isOpen');
    expect(isOpenState).toBeDefined();
  });

  test('have isOpenAdd state', () => {
    const wrapper = shallow(<App />);
    const isOpenAddState = wrapper.state('isOpenAdd');
    expect(isOpenAddState).toBeDefined();
  });

});

describe('test setState', () => {
  test('test setState of data', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ data: [] });
  });

  test('test setState of search', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ search: 'search test of setstate' });
  });
  
  test('test setState of id', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ id: 1 });
  });
  
  test('test setState of withTag', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ withTag: 'tagtest' });
  });
  
  test('test setState of  isOpen', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ isOpen: true });
  });

  test('test setState of isOpenAdd', () => {
    const wrapper = mount(<App />);
    wrapper.setState({ isOpenAdd: true });
  });
});
