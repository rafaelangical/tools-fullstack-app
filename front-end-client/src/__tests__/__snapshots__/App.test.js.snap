// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if App is rendered correctly renders correctly App 1`] = `
<div
  className="App"
>
  <main
    className="container"
  >
    <section
      className="div-top-title"
    >
      <h1>
        VUTTR
      </h1>
      <h3>
        Very Useful Tools to Remember
      </h3>
    </section>
    <section
      className="div-search-add"
    >
      <div
        className="div-search-add-left"
      >
        <input
          className="search"
          onChange={[Function]}
          placeholder="search"
          type="text"
          value=""
        />
        <input
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          search in tags only
        </label>
      </div>
      <div
        className="div-search-add-right"
      >
        <button
          onClick={[Function]}
        >
          + add
        </button>
      </div>
    </section>
    <section
      className="section-main"
    />
  </main>
</div>
`;
